### PR TITLE
post-comment

### PR DESCRIPTION
**Pull request by zaksaleh**

### CODE REVIEW

**Comment by Harrison365** [comment]
Hey Zak :) 

This is looking good so far 👍 However, there are a few things we need to include:

- Error handling if the user doesn't type anything in the box and tries to submit or if they haven't picked a user.
- You will also need error messages for unsuccessful posts (this can be achieved in a catch block).
- User shouldn't be able to post an empty comment.
- Form should clear after the post. 
- Should have a textarea input rather than an input tag so that they can write multiple lines. 
- The submit button should disable whilst post is happening. 

Apologies for the list. Let us know when you need us to take another look 👍

File: src/components/AddComment.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      <label>
+        <textarea
+          className="text-area"
+          rows="5"
+          cols="50"
+          placeholder="Add a comment..."
+          name="search"
+          value={body}
+          onChange={(event) => setBody(event.target.value)}
+        />
```
Comment (popatre) [inline src/components/AddComment.jsx:67]: Nice controlled component

File: src/components/AddComment.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      postComment(article_id, newComment)
+        .then((addedComment) => {
+          setComments((currentComments) => {
+            setButtonDisable(false);
+            setErr(null);
+            return [addedComment, ...currentComments];
+          });
+        })
+        .catch((err) => {
+          setErr("No Information! Please try again!");
```
Comment (popatre) [inline src/components/AddComment.jsx:36]: Only issue with this is i get the same message if i try and post offline. A bit confusing message - might just be better as `something went wrong`

**Review by popatre — CHANGES_REQUESTED** [review]
Just a small one to do with your error message. 

On a wider note, i dont think you have any loading things so when you first go on your pages if looks like they're broke. Would be good for some feedback to know something is happening, especially with render being quite slow sometimes.

**Pull request merged**